### ISSUE TITLE
API?

### ISSUE THREAD

**Issue by moolicc** [mikebryant/ac-nh-turnip-prices#33]
Disclosure: Webdev is a bit out of my field of expertise.

If there's interest for it and an API I can consume though, I can write a handy mobile app - just for the heck of it!

EDIT: I could always just write the prediction calculations myself, probably. But if somebody else has done that bit and got it working fine, I'd rather just collab with them on something than rewrite unnecessarily.

**Comment by theRTC204** [comment]
There is no backend for this, it's entirely on device. If you wanted to write a mobile app, you could write your own implementation of the algorithm in whatever language you'd like.

**Comment by Bazzatron19** [comment]
Would be really cool if the website supported URL parameters.
So if you navigated to:

`https://turnipprophet.io/index.html?buy=100&mam=10&mpm=20&tam=30&tpm=40`

it would automatically fill in the fields.

My particular use case is that I host a google doc with a lot of friends and family, everyone puts in there prices there, and we sort of work like a syndicate. More chances of profit.
Would be really neat if I could script a link that automatically populates the site without requiring a greasemonkey script to find the input elements and alter their values.

I'm really new to Git, so whilst I think I could add this functionality without too much issue, I don't know how to pull and commit etc.

**Comment by peter50216** [comment]
> Would be really cool if the website supported URL parameters.
> So if you navigated to:
> 
> `https://turnipprophet.io/index.html?buy=100&mam=10&mpm=20&tam=30&tpm=40`
> 
> it would automatically fill in the fields.
> 
> My particular use case is that I host a google doc with a lot of friends and family, everyone puts in there prices there, and we sort of work like a syndicate. More chances of profit.
> Would be really neat if I could script a link that automatically populates the site without requiring a greasemonkey script to find the input elements and alter their values.
> 
> I'm really new to Git, so whilst I think I could add this functionality without too much issue, I don't know how to pull and commit etc.

https://github.com/mikebryant/ac-nh-turnip-prices/pull/27 implements something similar to this with same use case :wink:

**Comment by Bazzatron19** [comment]
You're a gem Peter!

**Comment by theRTC204** [comment]
There's been a couple requests for this now, so definitely something worth considering.

Bigger concerns were around the loss of personal data because the app stores values in the browser's Local Storage.

I haven't looked back over @peter50216's PR since he adjusted for that concern, but I will do that today.

**Comment by moolicc** [comment]
Welp, if something like this gets implemented, I'll definitely consume it from a mobile app! :)

**Comment by theRTC204** [comment]
I'm likely going to re-implement the algorithm in Dart and use it to build out a mobile app in Flutter. If it doesn't take too long, and I like it, I'll probably release it.

**Comment by moolicc** [comment]
@theRTC204 sounds good!

I've started work on an implementation of the algorithm, but in kotlin. I've been looking for a project to learn kotlin with for a while now, so this might be perfect.

I've heard good things about dart and flutter though, I'd be very much interested in taking a peek at yours if you open source it 😉

**Comment by highb** [comment]
I wanted to learn about OpenAPI specs this weekend so I took a stab at making one and generating a server stub for it. 

I have it [running on Glitch](https://highb-stalks-as-a-service.glitch.me/?sellPrice=90) and the [code is here on Github](https://github.com/highb/stalks-as-a-service). It's probably riddled with bugs since I just forked `lib/prediction.js` from here without testing it out much, but it's a basic prototype. 😆
